src/ssl.ml

```diff
@@ -77,6 +77,26 @@ type verify_error =
   | Error_v_keyusage_no_certsign
   | Error_v_application_verification
 
+module Modes = struct
```

Comment:
alternative design:

- use a variant with all the modes 
- accept a list of modes
- `OR` the modes in the list to get mode

Reply:
Pro of the list design
- often use in OCAml
- allows for pattern matching
Cons
- slow
- allocates
- many more lines of codes.

Moreover, if OCaml allowed pattern matching on "constant", i.e. value known at compile time, pattern matching would still be possible.

Reply:
we can probably make it non-allocating if it's just pattern matches to convert to int, no?